internal/clock/simulated_clock_test.go

```diff
@@ -0,0 +1,287 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clock
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	// A non-zero reference time for tests
+	shortTestTimeout = 10 * time.Millisecond // For non-blocking channel checks
+	fireTestTimeout  = 50 * time.Millisecond // When expecting a channel to fire
+)
+
+func TestSimulatedClock_Now(t *testing.T) {
+	testCases := []struct {
+		name             string
+		initialTimeSetup func(sc *SimulatedClock)
+		expectedTime     time.Time
+	}{
+		{
+			name:             "InitialState_IsZeroTime",
+			initialTimeSetup: func(sc *SimulatedClock) {},
+			expectedTime:     time.Date(2020, time.January, 1, 12, 0, 0, 0, time.UTC),
+		},
+		{
+			name: "AfterSetTime_ReturnsSetTime",
+			initialTimeSetup: func(sc *SimulatedClock) {
+				sc.SetTime(time.Date(2020, time.January, 1, 12, 0, 0, 0, time.UTC))
+			},
+			expectedTime: time.Date(2020, time.January, 1, 12, 0, 0, 0, time.UTC),
+		},
+		{
+			name: "AfterAdvanceTime_ReturnsAdvancedTime",
+			initialTimeSetup: func(sc *SimulatedClock) {
+				sc.SetTime(time.Date(2020, time.January, 1, 12, 0, 0, 0, time.UTC))
+				sc.AdvanceTime(time.Hour)
+			},
+			expectedTime: time.Date(2020, time.January, 1, 13, 0, 0, 0, time.UTC),
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			clock := NewSimulatedClock(time.Date(2020, time.January, 1, 12, 0, 0, 0, time.UTC))
+			tc.initialTimeSetup(clock)
+
+			now := clock.Now()
+
+			assert.True(t, now.Equal(tc.expectedTime), "clock.Now() returned %v, expected %v", now, tc.expectedTime)
```

Comment:
Ditto elsewhere. Apologies for not catching this in the first pass.
```suggestion
			assert.Equal(t, tc.expectedTime), now)
```